contracts/FungibleToken-v2-ContractInterface.cdc

```diff
@@ -0,0 +1,34 @@
+import FungibleToken from "./FungibleToken-v2.cdc"
+
+pub contract interface FungibleTokenInterface {
+
+    /// TokensWithdrawn
+    ///
+    /// The event that is emitted when tokens are withdrawn from a Vault
+    pub event TokensWithdrawn(amount: UFix64, from: Address?, type: Type)
```

Comment:
>Are these events coming from the FT contract itself, though? For events, I feel that we want to be as explicit as possible because we do not know what other contracts will name themselves and they will all need to define these events individually (unless that is changing?). For instance, what if FiatToken had instead been named USDC? Now the event signature is just Withdrawn which isn't necessarily as clear anymore. It also helps differentiate from a common event name because Withdrawn is much more likely to be used elsewhere than TokensWithdrawn so any indexer trying to make sense of events is going to have less trouble

Events should always be standardised within the contract. If some FT is not implementing the standardized events, then those tokens are not compliant. Any other name is not compliant with FT standards. We can't restrict another contract to using the same event name, They can use the `TokensWithdrawn` or `Withdrawn`, but the context is automatically different as the contract address differs, so the indexer should not have a problem indexing the events.

Reply:
> Events should always be standardised within the contract. If some FT is not implementing the standardized events, then those tokens are not compliant. Any other name is not compliant with FT standards. We can't restrict another contract to using the same event name, They can use the TokensWithdrawn or Withdrawn, but the context is automatically different as the contract address differs, so the indexer should not have a problem indexing the events.

Not sure I follow, is the event now coming from the FungibleToken address? Otherwise I think I need more info about what you're saying.

Reply:
The reason I did it like this was because I wanted to keep the events as similar to their old versions as possible. Adding the argument onto the end would not break off-chain event listener code because it just adds another element to the array instead of replacing an existing element. I also would prefer to keep the same name for the same reason.